hydrolib/core/basemodel.py

```diff
@@ -1221,3 +1321,38 @@ def adjust_none(v: Any, field: ModelField) -> Any:
         return v
 
     return validator("*", allow_reuse=True, pre=True)(adjust_none)
+
+
+class UserInputValidation:
+    """Class to take care of user input validation."""
+
+    def __init__(self) -> None:
+        """Initializes a new instance of the `UserInputValidation` class."""
+        self._os_path_style = get_path_style_for_current_operating_system()
```

Comment:
I would say the `init` function is the best place. Because this is the initializer function for this object, which should initialize attributes and other stuff of this instance.
The `new` function is responsible for creating and returns an instance of the object type. 


>  [__new__()](https://docs.python.org/3/reference/datamodel.html?highlight=__new__#object.__new__) is intended mainly to allow subclasses of immutable types (like int, str, or tuple) to customize instance creation. It is also commonly overridden in custom metaclasses in order to customize class creation.

Reply:
Instead of instance variables, I made it static variables. So they are only initialized once during program execution. I decided not to make it a global variable.